core/process/status/exitstatus_spec.rb

```diff
@@ -10,4 +9,17 @@
     $?.exitstatus.should == 42
   end
 
+  describe "for a child that raised SignalException" do
+    before :each do
+      ruby_exe("raise SignalException, 'SIGTERM'")
+    end
+
+    platform_is_not :windows do
+      # The exitstatus is not set in these cases. See the termsig_spec
+      # for info on where the signal number (SIGTERM) is available.
+      it "returns nil" do
+        $?.exitstatus.should == nil
+      end
+    end
+  end
```

Comment:
I ended up adding this `exitstatus` spec also, since the behavior diverged there also between MRI and JRuby.